### DISCUSSION TITLE
How to list all periodic tasks

### DISCUSSION THREAD

**Discussion by RemiDesgrange** [celery/celery#8785] [Q&A]
I think everything is in the title.

I read https://docs.celeryq.dev/en/stable/userguide/periodic-tasks.html and https://docs.celeryq.dev/en/stable/userguide/monitoring.html 

`celery -A my_proj inspect scheduled` seems to be only for eta or countdown, it appears that there are no `inspect` commands  for beat (periodic tasks). 

am I missing something?

Thx.

**Comment by stevenparkerco3** [comment]
`celery -A app report` can show all app-configured variables in a handy report (including the `beat_schedule` you're after), though it's not a very good programmatic way to see the scheduled beat tasks.

The actual object you're after is the `app.conf.table()` dictionary, which defines a key called `beat_schedule`. Though, I don't believe the celery command line can effortlessly extract only this item.

For example:
```bash
celery -A app shell
>>> app.conf.table().get('beat_schedule')  # OR you can use app.conf.get('beat_schedule')
```

**Reply by RemiDesgrange** [reply]
I don't have any `beat_schedule` in the `table()`  function return, which seems to be a dump of django settings in my specific case. I'm using celery 5.3 BTW. Maybe a specific use case when django is used ?

**Reply by MajorDallas** [reply]
For others coming along in 2026, `app.conf.beat_schedule` is the current way to get the schedule dictionary.

**Reply by RemiDesgrange** [reply]
It lists all the periodic tasks. What I was hoping for was to be able to list when the next run for a task will happen.

**Comment by Ecno92** [comment]
@RemiDesgrange 
I am also using Django and did the following:

```python
from project.celery import app  # this is the app=Celery(...) instance
from celery.beat import Service
scheduler = Service(app).get_scheduler()
print(scheduler.get_schedule())
```

**Reply by RemiDesgrange** [reply]
I get a empty dict out of this in production env, while I can see execution in logs. I'm using redis as broker and result storage, maybe this feature works only with the django ORM broker ?

**Reply by Ecno92** [reply]
I don´t think that's the case. The project that I was working on is using a Django ORM based result backend and a redis broker.
